File: app/src/main/java/com/pirateman/exercisemanager/ExercisePlannerActivity.java
package com.pirateman.exercisemanager;

import android.annotation.TargetApi;
import android.databinding.ObservableArrayList;
import android.os.Build;
import android.support.annotation.RequiresApi;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.GridLayout;
import android.view.View;
import android.widget.CheckBox;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ThreadLocalRandom;

@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
public class ExercisePlannerActivity extends AppCompatActivity implements ActivityFragment.Callbacks {

    private  GridLayout layout;

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_exercise_planner);

        layout = findViewById(R.id.glExerciseGenerator);

        //maximum values
        String duration = getIntent().getStringExtra("duration");
        String incline = getIntent().getStringExtra("incline");
        String intervals = getIntent().getStringExtra("intervals");
        String speed = getIntent().getStringExtra("speed");

        int dur = Integer.valueOf(duration);
        double incl = Double.valueOf(incline);
        //will determine the number of intervals to create
        int inte = Integer.valueOf(intervals);
        double spe = Double.valueOf(speed);

        List<ExerciseInterval> exerciseIntervals = new ArrayList<>();

        for (int i = 0; i < inte; i++)
        {

            ExerciseInterval exerciseInterval = new ExerciseInterval(spe, incl, dur);
            addFragment(exerciseInterval);
        }
//
//        List<ExerciseInterval> exerciseIntervals = new ArrayList<>();
//        exerciseIntervals.add(new ExerciseInterval(0, 3.3, 1.0, 2));
//        exerciseIntervals.add(new ExerciseInterval(0, 5.3, 2.0, 3));
//        exerciseIntervals.add(new ExerciseInterval(0, 1.0, 1.0, 2));
//        exerciseIntervals.add(new ExerciseInterval(0, 3.3, 2.5, 1));
//
//        for (ExerciseInterval interval : exerciseIntervals)
//        {
//
//            addFragment(interval);
//
////            addTextView(interval.getActivity(), interval.getRowId(), 0);
////            addTextView(String.valueOf(interval.getSpeed()), interval.getRowId(), 1);
////            addTextView(String.valueOf(interval.getIncline()), interval.getRowId(), 2);
////            addTextView(String.valueOf(interval.getDuration()), interval.getRowId(), 3);
//        }

    }

    public void addFragment(ExerciseInterval interval)
    {
        ActivityFragment fragment = new ActivityFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable("exerciseInterval", interval);
        fragment.setArguments(bundle);

        getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.flContainer, fragment)
                .addToBackStack(null)
                .commit();
    }

    public void replaceFragment(ExerciseInterval interval)
    {
        ActivityFragment fragment = new ActivityFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable("exerciseInterval", interval);
        fragment.setArguments(bundle);

//        ObservableArrayList<Object> user = new ObservableArrayList<>();
//        user.add("Google");
//        user.add("Inc.");
//        user.add(17);

        getSupportFragmentManager()
                .beginTransaction()

                .replace(R.id.flContainer, fragment)
                .commit();
    }

    public void addTextView(String str, int row, int col)
    {
        TextView textView = new TextView(this);

        textView.setText(str);
        textView.setTextSize(12f);

        GridLayout.Spec rowSpan = GridLayout.spec(row, 1);
        GridLayout.Spec colspan = GridLayout.spec(col, 1);

        GridLayout.LayoutParams gridParam = new GridLayout.LayoutParams(
                rowSpan, colspan);
        layout.addView(textView, gridParam);

    }

//    public void addTextView(String str, int row, int col)
//    {
//        TextView textView = new TextView(this);
//
//        textView.setText(str);
//        textView.setTextSize(12f);
//
//        GridLayout.Spec rowSpan = GridLayout.spec(row, 1);
//        GridLayout.Spec colspan = GridLayout.spec(col, 1);
//
//        GridLayout.LayoutParams gridParam = new GridLayout.LayoutParams(
//                rowSpan, colspan);
//        layout.addView(textView, gridParam);
//
//    }

    public void addCheckBox(int row, int col)
    {
        CheckBox checkBox = new CheckBox(this);

        GridLayout.Spec rowSpan = GridLayout.spec(row, 1);
        GridLayout.Spec colspan = GridLayout.spec(col, 1);

        GridLayout.LayoutParams gridParam = new GridLayout.LayoutParams(
                rowSpan, colspan);
        layout.addView(checkBox, gridParam);

    }


    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    public void onReplaceFragmentClick(View v)
    {
        ExerciseInterval newInterval = new ExerciseInterval();
        replaceFragment(newInterval);
    }

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    public void onAddFragmentClick(View v)
    {
        ExerciseInterval newInterval = new ExerciseInterval();
        addFragment(newInterval);
    }


    @Override
    public void onButtonClicked()
    {
        replaceFragment(new ExerciseInterval());
    }
}

File: app/src/main/java/com/pirateman/exercisemanager/ExercisePlannerActivity.kt
package com.pirateman.exercisemanager

import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.CheckBox
import android.widget.GridLayout
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import java.util.*

@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
class ExercisePlannerActivity : AppCompatActivity(), ActivityFragment.Callbacks {
    private var layout: GridLayout? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_exercise_planner)
        layout = findViewById(R.id.glExerciseGenerator)

        //maximum values
        val duration = intent.getStringExtra("duration")
        val incline = intent.getStringExtra("incline")
        val intervals = intent.getStringExtra("intervals")
        val speed = intent.getStringExtra("speed")
        val dur = Integer.valueOf(duration)
        val incl = java.lang.Double.valueOf(incline)
        //will determine the number of intervals to create
        val inte = Integer.valueOf(intervals)
        val spe = java.lang.Double.valueOf(speed)
        val exerciseIntervals: List<ExerciseInterval> = ArrayList()
        for (i in 0 until inte) {
            val exerciseInterval = ExerciseInterval(spe, incl, dur)
            addFragment(exerciseInterval)
        }
        //
//        List<ExerciseInterval> exerciseIntervals = new ArrayList<>();
//        exerciseIntervals.add(new ExerciseInterval(0, 3.3, 1.0, 2));
//        exerciseIntervals.add(new ExerciseInterval(0, 5.3, 2.0, 3));
//        exerciseIntervals.add(new ExerciseInterval(0, 1.0, 1.0, 2));
//        exerciseIntervals.add(new ExerciseInterval(0, 3.3, 2.5, 1));
//
//        for (ExerciseInterval interval : exerciseIntervals)
//        {
//
//            addFragment(interval);
//
////            addTextView(interval.getActivity(), interval.getRowId(), 0);
////            addTextView(String.valueOf(interval.getSpeed()), interval.getRowId(), 1);
////            addTextView(String.valueOf(interval.getIncline()), interval.getRowId(), 2);
////            addTextView(String.valueOf(interval.getDuration()), interval.getRowId(), 3);
//        }
    }

    fun addFragment(interval: ExerciseInterval?) {
        val fragment = ActivityFragment()
        val bundle = Bundle()
        bundle.putSerializable("exerciseInterval", interval)
        fragment.arguments = bundle
        supportFragmentManager
                .beginTransaction()
                .add(R.id.flContainer, fragment)
                .addToBackStack(null)
                .commit()
    }

    fun replaceFragment(interval: ExerciseInterval?) {
        val fragment = ActivityFragment()
        val bundle = Bundle()
        bundle.putSerializable("exerciseInterval", interval)
        fragment.arguments = bundle

//        ObservableArrayList<Object> user = new ObservableArrayList<>();
//        user.add("Google");
//        user.add("Inc.");
//        user.add(17);
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.flContainer, fragment)
                .commit()
    }

    fun addTextView(str: String?, row: Int, col: Int) {
        val textView = TextView(this)
        textView.text = str
        textView.textSize = 12f
        val rowSpan = GridLayout.spec(row, 1)
        val colspan = GridLayout.spec(col, 1)
        val gridParam = GridLayout.LayoutParams(
                rowSpan, colspan)
        layout!!.addView(textView, gridParam)
    }

    //    public void addTextView(String str, int row, int col)
    //    {
    //        TextView textView = new TextView(this);
    //
    //        textView.setText(str);
    //        textView.setTextSize(12f);
    //
    //        GridLayout.Spec rowSpan = GridLayout.spec(row, 1);
    //        GridLayout.Spec colspan = GridLayout.spec(col, 1);
    //
    //        GridLayout.LayoutParams gridParam = new GridLayout.LayoutParams(
    //                rowSpan, colspan);
    //        layout.addView(textView, gridParam);
    //
    //    }
    fun addCheckBox(row: Int, col: Int) {
        val checkBox = CheckBox(this)
        val rowSpan = GridLayout.spec(row, 1)
        val colspan = GridLayout.spec(col, 1)
        val gridParam = GridLayout.LayoutParams(
                rowSpan, colspan)
        layout!!.addView(checkBox, gridParam)
    }

    fun onReplaceFragmentClick(v: View?) {
        val newInterval = ExerciseInterval()
        replaceFragment(newInterval)
    }

    fun onAddFragmentClick(v: View?) {
        val newInterval = ExerciseInterval()
        addFragment(newInterval)
    }

    override fun onButtonClicked() {
        replaceFragment(ExerciseInterval())
    }
}